### PR TITLE
Default assembler refactor

### PR DESCRIPTION
**Pull request by pith**
- Rename `AutomaticAssembler` in `ModelMapperAssembler`
- Default assemblers are now injected with `Assembler<?,?>`
  and a qualifier.
- Additionnal _default assemblers_ can be created via the SPI
- The DSL has been moved from core to internal package

### CODE REVIEW

File: core/src/it/java/org/seedstack/business/assembler/auto/fixture/DummyDefaultAssembler.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * @author pierre.thirouin@ext.mpsa.com (Pierre Thirouin)
+ */
+@DefaultImpl
+@Named("Dummy")
+public class DummyDefaultAssembler<A extends AggregateRoot<?>,D> extends AbstractBaseAssembler<A, D> {
+
+
+    @SuppressWarnings("unchecked")
+    @Inject
+    public DummyDefaultAssembler(@Assisted Object[] genericClasses) {
```
Comment (ejemba) [inline core/src/it/java/org/seedstack/business/assembler/auto/fixture/DummyDefaultAssembler.java:31]: Do projects have to use `@Assisted` which is a guice specific class or this is just for the tests ?

File: core/src/test/java/org/seedstack/business/internal/interfaces/assembler/dsl/AssembleTest.java [hunk 2]
```diff
@@ -1,95 +1,95 @@
...
+    private Order order2 = new Order();
+    private List<Object> dtos = Lists.newArrayList(orderDto, myOrder);
+    private List<Order> orders = Lists.newArrayList(myOrder, myOrder);
+
+    private FluentAssembler fluently;
+
+    public void testDsl() {
+
+        // dto to aggregate
+        order = fluently.assemble().dto(orderDto).to(myOrder);
```
Comment (ejemba) [inline core/src/test/java/org/seedstack/business/internal/interfaces/assembler/dsl/AssembleTest.java:43]: I must admit I don't understand the `.to(myOrder)` . This should be a class , isn't it ? If no what the difference between `order` and `myOrder` ?

File: core/src/test/java/org/seedstack/business/internal/interfaces/assembler/dsl/AssembleTest.java [hunk 3]
```diff
@@ -1,95 +1,95 @@
...
+
+        // from factory
+        order = fluently.assemble().dto(orderDto).to(Order.class).fromFactory();
+
+        // list of dto to tuple of aggregates
+        Pair<Order, Customer> orderCustomerPair = fluently.assemble().dto(orderDto).to(Order.class, Customer.class).fromFactory();
+
+        // list of dtos to list of aggregates
+        orders = fluently.assemble().dtos(dtos).to(orders);
```
Comment (ejemba) [inline core/src/test/java/org/seedstack/business/internal/interfaces/assembler/dsl/AssembleTest.java:52]: Same remark. This is not clear to me. why two `orders` here ?

**Comment by ejemba** [comment]
Huge refactor ! congrats !
@pith can you just answer the question I add ?

File: core/src/it/java/org/seedstack/business/assembler/auto/fixture/DummyDefaultAssembler.java (hunk 1 above)
Comment (pith) [inline core/src/it/java/org/seedstack/business/assembler/auto/fixture/DummyDefaultAssembler.java:31]: No assembler users don't have to use it. It is only required for developers who wants to use the SPI for creating new _default assemblers_. It is mandatory for getting the DTO class using the _binding strategy_.

File: core/src/test/java/org/seedstack/business/internal/interfaces/assembler/dsl/AssembleTest.java (hunk 2 above)
Comment (pith) [inline core/src/test/java/org/seedstack/business/internal/interfaces/assembler/dsl/AssembleTest.java:43]: There is two cases:
- assembling to a aggregate root coming from a factory or a repository. In this case a class is passed to the `to()` method.
- assembling to a given aggregate root. Here an instance is passed.

In the case of this test, `myOrder` and `order` are equals (same aggregate root) but `order` is updated with the data coming from `dto`.

**Comment by pith** [comment]
@ejemba thanks and done ;).

File: core/src/test/java/org/seedstack/business/internal/interfaces/assembler/dsl/AssembleTest.java (hunk 2 above)
Comment (ejemba) [inline core/src/test/java/org/seedstack/business/internal/interfaces/assembler/dsl/AssembleTest.java:43]: Isn't this confusing ? 
The semantic used in Assembler is not respected that is why I feel.
1. Assemble a dto from an aggregate
2. Merge an aggregate from a dto.

I mean we won't assemble an aggregate, but merge data into it.
For this use I was expecting something like
`fluently.merge().dto(orderDto).into(myOrder);`
Merge indicates we will merge data into `myOrder`. 

I really think the previous semantic should be kept for consistency.

Comment (pith) [inline core/src/test/java/org/seedstack/business/internal/interfaces/assembler/dsl/AssembleTest.java:43]: This will be fixed here #31

**Pull request merged**